pkgs/tools/filesystems/bees/bees-service-wrapper

```diff
@@ -0,0 +1,197 @@
+  [idxSizeMB]=bees_db_size_mb
+  [verbosity]=bees_verbosity
+  [workDir]=bees_work_dir
+)
+
+# legacy bees config files can be arbitrary shell scripts, so we need to actually evaluate them
+readConfigFileIfExists() {
+  [[ -s $1 ]] || return 1
+  source "$1"
+  for oldName in "${!configNames[@]}"; do : "$oldName"
+    if [[ ${!oldName} ]]; then
+      newName=${configNames[$oldName]}
+      printf -v "$newName" %s "${!oldName}"
+      unset "$oldName"
+    fi
+  done
+}
+
+uuid_re='^[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}$'
+
+# Shared code for setting configuration used by other operations.
+#
+# Reads from global associative array "options" containing options passed in as
+# key=value pairs on the command line, looks for config-file overrides, and
+# sets individual global variables.
+_setup() {
+  declare fstype fs_spec=$1; shift
+  declare -g bees_{blockdev,uuid,verbosity} bees_{work,run,mnt}_dir bees_db_size{,_mb}
+  for key in "${!options[@]}"; do
+    # Rename options as appropriate
+    if [[ ${configNames[$key]} && ${configNames[$key]} != "$key" ]]; then
+      options[${configNames[$key]}]=${options[$key]}
+      unset "options[$key]"
+    fi
+  done
+
+  # Look for file-based configuration, additional to honoring configuration on the command line
+  bees_config_dir="${bees_config_dir:-/etc/bees}"
+  if [[ $fs_spec =~ $uuid_re ]]; then
+    bees_uuid=$fs_spec
+    # If our spec looks like a bare UUID, and no config file exists in the new
+    # format, fall back to legacy config file search mechanism (grep; ewww).
+    if ! readConfigFileIfExists "$bees_config_dir/UUID=$fs_spec.conf"; then
+      # Legacy approach to finding a config file: Grep for a *.conf file
+      # containing the UUID within its text. Permitting spaces around the "="
+      # appears to be a bug, but is retained for compatibility with the
+      # original upstream script.
+      allConfFiles=( "$bees_config_dir"/*.conf )
+      if (( ${#allConfFiles[@]} )); then
+        readarray -d '' -t matchingConfFiles < <(grep -E -l -Z "^[^#]*UUID[[:space:]]*=[[:space:]]*" "$bees_config_dir"/*.conf)
+      else
+        matchingConfFiles=( )
+      fi
+      if (( ${#matchingConfFiles[@]} == 1 )); then
+        # Exactly one configuration file exists in our target directory with a reference to the UUID given.
+        bees_config_file=${matchingConfFiles[0]}
+        readConfigFileIfExists "$bees_config_file"
+        echo "NOTE: Please consider renaming $bees_config_file to $bees_config_dir/UUID=$fs_spec" >&2
+        echo "      ...and passing UUID=$fs_spec on startup." >&2
+      elif (( ${#matchingConfFiles[@]} > 1 )); then
+        # The legacy wrapper would silently use the first file and ignore
+        # others, but... no.
+        echo "ERROR: Passed a bare UUID, but multiple configuration files match it:" >&2
+        printf ' - %q\n' "${matchingConfFiles[@]}" >&2
+        echo "Unable to continue." >&2
+        exit 1
```

Comment:
This is exactly what `die "Unable to continue."`  would do, right?

Reply:
Correct.